user_tools/src/spark_rapids_tools/tools/qualx/hash_util.py

```diff
@@ -116,37 +114,57 @@ def path_match(node, expected_path: list[str]):
         if node['nodeName'] == expected_path[0]:
             if len(expected_path) == 1:
                 return True
-        if len(node['children']) == 1:
-            return path_match(node['children'][0], expected_path[1:])
+            return any(path_match(child, expected_path[1:]) for child in node['children'])
     return False
 
 
 def normalize_plan(plan):
     """Normalize the sparkPlanInfo for comparing CPU and GPU plans.
 
-    The high-level algorithm is a two-step process:
-    1. Traverse the plan tree to normalize each node.
+    The high-level algorithm is a multi-step process:
+    1. Traverse the plan tree to normalize individual nodes.
       - Remove any nodes in the remove_nodes_exact list
       - Remove any nodes by prefix match to the remove_nodes_prefix list
       - Rename nodes to a "canonical" form using the rename_nodes dictionary
-    2. Traverse the plan tree again to normalize entire paths (sequences of nodes).
+      - Remove any SubqueryAdaptiveBroadcast children
+    2. Traverse the plan tree to normalize entire paths (sequences of nodes).
       - GpuTopN/GpuTopN/X -> TakeOrderedAndProject/X
       - Project/Filter/X -> Project/X
       - UnionWithLocalData/Union/X -> X
-      - X/SubqueryAdaptiveBroadcast -> X
+    3. Traverse the plan tree to normalize BroadcastHashJoin subtrees.
+    4. Traverse the plan tree to normalize SortMergeJoin/Project/BroadcastHashJoin subtrees.
 
     Notes:
     - The normalization should be applied to both CPU and GPU plans.
-    - This does not handle cases where the GPU plan has been significantly modified.
     - Eventlogs only contain physical plans, so use the earliest (least modified) physical plan.
     - The goal is to produce a "signature" of the plan and not a faithful re-creation of the logical plan.
     """
 
+    def remove_nodes(node, remove_list) -> List[dict]:
+        """Remove any nodes in the remove_list from a plan tree, while preserving children."""
+        children = []
+        for child in node['children']:
+            children.extend(remove_nodes(child, remove_list))
+
+        if node['nodeName'] in remove_list:
+            return children
+
+        node['children'] = children
+        return [node]
+
     def normalize_node(node):
-        """Normalize a single node in the plan by removing and/or renaming to canonical form."""
+        """Normalize a single node in the plan by removing and/or renaming to canonical form.
+
+        Notes:
+        - SubqueryAdaptiveBroadcast children (and their subtrees) are also removed.
+        """
         if isinstance(node, dict):
             if 'nodeName' in node:
                 node_name = node['nodeName']
+                # remove SubqueryAdaptiveBroadcast children
+                node['children'] = [
+                    child for child in node['children'] if child['nodeName'] != 'SubqueryAdaptiveBroadcast'
+                ]
```

Comment:
Do you think we should also clear that out from the qual execs and unsupported_ops report? I remember there is an existing issue to remove those because it confuses the users.